### PR TITLE
Avoid terms "Primary/Secondary Resource"

### PR DESCRIPTION
**Pull request by peacekeeper**
This PR addresses https://github.com/w3c/did-resolution/issues/94 and replaces https://github.com/w3c/did-resolution/pull/89.

It now has section headings called "Dereferencing the Resource" and "Dereferencing the Fragment" instead of using "Primary Resource" and "Secondary Resource" (except in a small note that references RFC3986).

### CODE REVIEW

**Review by msporny — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -1108,8 +1106,8 @@ <h2>Dereferencing the Secondary Resource</h2>
 					<code>Location</code> header (see section 7.1.2 of [[RFC7231]].
 					</p>
 				</ol>
-				<li>Otherwise, dereference the secondary resource identified by the DID fragment as defined by the media type ([[RFC2046]]) of the primary resource.
-				For example, if the primary resource is a representation of a DID document with media type <code>application/did+ld+json</code>, then
+				<li>Otherwise, dereference the DID fragment as defined by the media type ([[RFC2046]]) of the resource.
+				For example, if the resource is a representation of a DID document with media type <code>application/did+ld+json</code>, then
```
Comment (msporny) [inline index.html:1110]:
```suggestion
				For example, if the resource is a representation of a DID document with media type <code>application/did</code>, then
```

File: index.html [hunk 2]
```diff
@@ -1001,36 +1001,41 @@ <h3>DID URL Dereferencing Metadata</h3>
 	<section id="dereferencing-algorithm">
 		<h2>Algorithm</h2>
 		<p>The following <a>DID URL dereferencing</a> algorithm MUST be implemented by a conformant <a>DID resolver</a>.
-		In accordance with [[RFC3986]], it consists of the following steps: Resolving the DID, dereferencing the primary
-		resource, and dereferencing the secondary resource (only if the <var>input <a>DID URL</a></var> contains a <a>DID fragment</a>:</p>
+		In accordance with [[RFC3986]], it consists of the following steps: Resolving the DID, dereferencing the
+		resource, and dereferencing the fragment (only if the <var>input <a>DID URL</a></var> contains a <a>DID fragment</a>):</p>
```
Comment (TallTed) [inline index.html:1005]:
```suggestion
		In accordance with [[RFC3986]], it consists of the following three steps: resolving the DID; dereferencing the
		resource; and dereferencing the fragment (only if the <var>input <a>DID URL</a></var> contains a <a>DID fragment</a>):</p>
```

File: index.html [hunk 3]
```diff
@@ -1001,36 +1001,41 @@ <h3>DID URL Dereferencing Metadata</h3>
...
+				<li>Obtain the <a>DID document</a> for the <var>input <a>DID</a></var> by executing the
+					<a>DID resolution</a> algorithm as defined in <a href="#resolving"></a>. All
+					<a href="https://www.w3.org/TR/did-core/#did-parameters">
+						DID parameters</a> of the <var>input <a>DID URL</a></var> MUST be passed as <var>resolution options</var> to the <a>DID
+						Resolution</a> algorithm. If the <var>input <a>DID</a></var> does not exist, return a null result.
+					Otherwise, the result is called the <var>resolved <a>DID document</a></var>.</li>
+				</li>
+
+				<li>If present, separate the <a>DID fragment</a> from the <var>input <a>DID URL</a></var> and continue
+					with the <var>input <a>DID URL</a></var> adjusted accordingly.</li>
```
Comment (TallTed) [inline index.html:1029]:
```suggestion
					with the adjusted <var>input <a>DID URL</a></var>.</li>
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Small stuff. Editorial.

**Review by danpape — APPROVED** [review]
Looks good.

**Comment by peacekeeper** [comment]
Merging after several approvals, suggestions applied, and merge conflicts resolved.

**Pull request merged**